AnkiHubSoftware/ankihub_addon
# feat: Hide "Submit without review" checkbox on suggestion dialog if user isn't authorized to use it

This PR makes some improvements to the `SuggestionDialog`.
We will hide the “Submit without review” check box for the standard user. For maintainers and owners we will keep the same behavior.
There will be no need for the alert dialog shown on the screenshot.
<img src="https://github.com/ankipalace/ankihub_addon/assets/31575114/03608525-9255-4b75-99e8-bc6b2e9fa570" width="700" />

## Related issues
[Improve the suggestions dialog ](https://www.notion.so/ankihub/AnkiHub-Planning-7d2bdb8e15034c559bef129adbe163b7?p=06bc30dd25114298a31f147332035f72&pm=c)

## Proposed changes
- Only show the checkbox if the user is a deck maintainer or owner
- Update tests

## How to reproduce
- Install one for which you are the owner
- Try to create a change note suggestion for a note in the deck
- The checkbox should be shown

- Install a deck for which you are not an owner or maintainer
- Try to create a change note suggestion for a note in the deck
- The checkbox should be hidden

## Screenshots and videos
I'm only a subscriber of the **Physics** deck, so the checkbox doesn't show up when I open the suggestion dialog.
I'm the owner of the **basic notes** deck, so the checkbox does show up.

https://github.com/ankipalace/ankihub_addon/assets/31575114/37768eae-7ec5-4c09-b950-400d7061a67c
